test/UnitTests/MSTest.Analyzers.UnitTests/AssertionArgsShouldAvoidConditionalAccessAnalyzerTests.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#nullable enable
+
+using Microsoft.Testing.Internal.Framework;
+using Microsoft.Testing.TestInfrastructure;
+
+using VerifyCS = MSTest.Analyzers.Test.CSharpCodeFixVerifier<
+    MSTest.Analyzers.AssertionArgsShouldAvoidConditionalAccessAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace MSTest.Analyzers.UnitTests;
+
+[TestGroup]
+public sealed class AssertionArgsShouldAvoidConditionalAccessAnalyzerTests(ITestExecutionContext testExecutionContext) : TestBase(testExecutionContext)
+{
+    public async Task WhenUsingConditionalsAccess_In_Assert_Equal()
+    {
+        string code = """
+            #nullable enable
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            using System.Collections.Generic;
+
+            public class A
+            {
+                public string? S { get; set; }
+                public List<string> T { get; set; } = new();
+            }
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [TestMethod]
+                public void NonCompliant()
```

Comment:
You are missing `AreSame` and `AreNotSame` cases.

Reply:
Added